test/asm/for.err

```diff
@@ -1,6 +1,6 @@
 ERROR: for.asm(16):
     FOR cannot have a step value of 0
-ERROR: for.asm(41) -> for.asm::REPT~5(47):
+ERROR: for.asm(41) -> for.asm::REPT~4(47):
     'v' already defined as constant at for.asm(41) -> for.asm::REPT~4(45)
-FATAL: for.asm(41) -> for.asm::REPT~5(47):
+FATAL: for.asm(41) -> for.asm::REPT~4(47):
```

Comment:
Why did this change?

Reply:
The iteration counter was being increased before attempting to update the for loop's value. This way, the error message reports the iteration in which the update actually failed.